zookeeper/init.sls

```diff
@@ -27,17 +27,30 @@ install-zookeeper-dist:
 {%- else %}
     - skip_verify: True
 {%- endif %}
+    - unless: test -d {{ zk.real_home }}/lib
   cmd.run:
     - name: tar xzf /usr/local/src/{{ zk.version_name }}.tar.gz --no-same-owner
     - cwd: {{ zk.prefix }}
-    - unless: test -d {{ zk.real_home }}/lib
+    - unless: test -d {{ zk.real_home }}/lib || ! test -f /usr/local/src/{{ zk.version_name }}.tar.gz
```

Comment:
This looks too complicated. Would you mind to replace 3 states (`file: install-zookeeper-dist`, `cmd: install-zookeeper-dist` and `file: remove_zookeeper_tar` ) with single `archive.extracted` function?